### ISSUE TITLE
validate API. Safe validate against multiple validators

### ISSUE THREAD

**Issue by o-alexandrov** [typeofweb/schema#51]
## Problem

Currently, it is impossible to safely validate against multiple validators.

## Proposed solution

```ts
export const validateSafe = <V, Validators extends ((v) => any)[]>(
  value: V,
  validators: Validators,
) => {
  let result: V | undefined
  for (const validator of validators) {
    try {
      result = validator(value)
    } catch {}
  }

  return result
}

const body = validateSafe(value, [validator1, validator2])
```

**Comment by mmiszy** [comment]
What's the use-case for this @o-alexandrov ?

**Comment by o-alexandrov** [comment]
@mmiszy to be honest, when you asked about the use case, I realized that I don't need a new helper to not throw an error in my own case. The reason why I used it is because I was migrating from [zod's safeParse](https://github.com/colinhacks/zod#safe-parse) and there, `.safeParse` was the only way it would have worked there.

With `@typeofweb/schema` since validators & schemas are separate, I could have used [oneOf](https://schema.typeofweb.com/docs/validators/oneOf) for the validator to accept one of the schemas.

So I'm sorry for wasting your time and I'll be closing this issue.
If you believe you'd need [zod's safeParse](https://github.com/colinhacks/zod#safe-parse), you are welcome to reopen.
- I don't see a need to validate without throwing to be popular enough to complicate this project
